test-clients/src/main/java/com/hedera/services/bdd/suites/token/UniqueTokenManagementSpecs.java

```diff
@@ -0,0 +1,322 @@
+import static com.hedera.services.bdd.spec.utilops.UtilVerbs.newKeyNamed;
+import static com.hederahashgraph.api.proto.java.ResponseCodeEnum.FAIL_INVALID;
+import static com.hederahashgraph.api.proto.java.ResponseCodeEnum.INVALID_NFT_ID;
+import static com.hederahashgraph.api.proto.java.ResponseCodeEnum.INVALID_TOKEN_NFT_SERIAL_NUMBER;
+import static com.hederahashgraph.api.proto.java.ResponseCodeEnum.INVALID_TRANSACTION_BODY;
+import static com.hederahashgraph.api.proto.java.ResponseCodeEnum.SUCCESS;
+import static com.hederahashgraph.api.proto.java.ResponseCodeEnum.TOKEN_WAS_DELETED;
+
+public class UniqueTokenManagementSpecs extends HapiApiSuite {
+	private static final org.apache.logging.log4j.Logger log = LogManager.getLogger(UniqueTokenManagementSpecs.class);
+	private static final String TOKEN_NAME = "token";
+	private static final String SUPPLY_KEY = "supplyKey";
+	public static void main(String... args) {
+		new UniqueTokenManagementSpecs().runSuiteSync();
+	}
+
+	@Override
+	protected List<HapiApiSpec> getSpecsInSuite() {
+		return List.of(new HapiApiSpec[]{
+						getTokenNftInfoWork(),
+						uniqueTokenHappyPath(),
+						sadPathWithFrozenToken(),
+						sadPathWithDeletedToken(),
+						sadPathWithRepeatedMetadata(),
+						happyPathCallingMintOnceWithFiveMetadata(),
+						happyPathCallMintFiveTimesWithOneMetadata(),
+				}
+		);
+	}
+
+
+	private HapiApiSpec uniqueTokenHappyPath() {
+		return defaultHapiSpec("UniqueTokenHappyPath")
+				.given(
+						newKeyNamed(SUPPLY_KEY),
+						cryptoCreate(TOKEN_TREASURY),
+						tokenCreate(TOKEN_NAME)
+								.tokenType(TokenType.NON_FUNGIBLE_UNIQUE)
+								.supplyType(TokenSupplyType.INFINITE)
+								.initialSupply(0)
+								.supplyKey(SUPPLY_KEY)
+								.treasury(TOKEN_TREASURY)
+				).when(
+						mintToken(TOKEN_NAME, List.of(ByteString.copyFromUtf8("memo"), ByteString.copyFromUtf8("memo1"))).via("mintTxn")
+				).then(
+						getReceipt("mintTxn")
+								.hasPriorityStatus(SUCCESS)
+								.hasCostAnswerPrecheck(SUCCESS),
+
+						getTokenNftInfo(TOKEN_NAME, 1)
+								.hasSerialNum(1)
+								.hasMetadata(ByteString.copyFromUtf8("memo"))
+								.hasTokenID(TOKEN_NAME)
+								.hasAccountID(TOKEN_TREASURY),
+
+						getTokenNftInfo(TOKEN_NAME, 2)
+								.hasSerialNum(2)
+								.hasMetadata(ByteString.copyFromUtf8("memo1"))
+								.hasTokenID(TOKEN_NAME)
+								.hasAccountID(TOKEN_TREASURY),
+
+						getTokenNftInfo(TOKEN_NAME, 3)
+								.hasCostAnswerPrecheck(INVALID_NFT_ID),
+
+						getAccountBalance(TOKEN_TREASURY)
+								.hasTokenBalance(TOKEN_NAME, 2),
+						getTokenInfo(TOKEN_NAME)
+								.hasTreasury(TOKEN_TREASURY)
+								.hasTotalSupply(0),
+
+						getAccountInfo(TOKEN_TREASURY)
+								.hasToken(relationshipWith(TOKEN_NAME))
+				);
+	}
+
+	private HapiApiSpec happyPathCallingMintOnceWithFiveMetadata() {
+		return defaultHapiSpec("happyPathCallingMintOnceWithFiveMetadata")
```

Comment:
`happyPathCallingMintOnceWithFiveMetadata` this looks and sounds awkward.
I suggest `happyPathWithBatchMetadata`